app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,21 @@
+class PiecesController < ApplicationController
+  
+  def update
+    if selected_piece.valid_move?(params[:piece][:x_position].to_i,params[:piece][:y_position].to_i)
+      selected_piece.update_attributes(piece_params)
+      render json: { response: "Valid Move"}
```

Comment:
yeah, if i take out the render json part it throws a missing template error. still trying to figure out how to get that fixed.